src/librustc_mir/borrow_check/places_conflict.rs

```diff
@@ -170,15 +170,6 @@ fn place_components_conflict<'gcx, 'tcx>(
                         debug!("places_conflict: shallow access behind ptr");
                         return false;
                     }
-                    (ProjectionElem::Deref, ty::Ref(_, _, hir::MutImmutable), _) => {
-                        // the borrow goes through a dereference of a shared reference.
-                        //
-                        // I'm not sure why we are tracking these borrows - shared
-                        // references can *always* be aliased, which means the
-                        // permission check already account for this borrow.
-                        debug!("places_conflict: behind a shared ref");
-                        return false;
-                    }
```

Comment:
As I wrote in a comment somewhere else:

(a) I think that the second element of the tuples here (the type) is now unused
(b) Why is it ok to remove this logic -- is it because we aren't tracking the borrows anymore? If so, maybe we can add an assertion of some kind...